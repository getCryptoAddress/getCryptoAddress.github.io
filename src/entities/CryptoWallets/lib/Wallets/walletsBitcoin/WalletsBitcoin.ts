import WalletsAbstract from "@/entities/CryptoWallets/lib/Wallets/WalletsAbstract";
import type Wallet from "@/entities/CryptoWallets/lib/Wallets/useWallet.types";
import generatePrivateKey from "@/entities/CryptoWallets/lib/PrivateKeys/generatePrivateKey";
import { getAddress, NETWORK, TEST_NETWORK } from "micro-btc-signer";
import convertUint8ArrayToHex from "@/entities/CryptoWallets/lib/PrivateKeys/convertUint8ArrayToHex";
import convertUint8ArrayToWif from "@/entities/CryptoWallets/lib/PrivateKeys/convertUint8ArrayToWif";
import type {
  BitcoinPrivateKey,
  BitcoinPrivateKeyFormat,
  BitcoinWalletPayload,
} from "./WalletsBitcoin.types";
import validatePrivateKey from "@/entities/CryptoWallets/lib/PrivateKeys/validatePrivateKey";

export default class WalletsBitcoin extends WalletsAbstract<
  BitcoinWalletPayload,
  BitcoinPrivateKey
> {
  async makeWallet(
    { formatPrivateKey, formatAddress, isTestnet }: BitcoinWalletPayload,
    initialPrivateKey?: BitcoinPrivateKey
  ): Promise<Wallet> {
    const nextPrivateKey = await this.#getPrivateKey(initialPrivateKey);

    const network = isTestnet ? TEST_NETWORK : NETWORK;
    const address = getAddress(formatAddress, nextPrivateKey, network);
    if (!address) {
      throw new Error("Invalid address");
    }
    const privateKey = await this.#getFormattedPrivateKey(
      nextPrivateKey,
      formatPrivateKey
    );

    return {
      privateKey,
      address,
    };
  }

  async #getPrivateKey(initialPrivateKey?: BitcoinPrivateKey) {
    if (initialPrivateKey) {
      if (!(await validatePrivateKey(initialPrivateKey, 32))) {
        throw new Error("Invalid private key");
      }
      return initialPrivateKey;
    }
    return generatePrivateKey(32);
  }

  async #getFormattedPrivateKey(
    key: Uint8Array,
    format: BitcoinPrivateKeyFormat
  ): Promise<string> {
    if (format === "hex") {
      return convertUint8ArrayToHex(key);
    }
    return convertUint8ArrayToWif(key, format);
  }
}
