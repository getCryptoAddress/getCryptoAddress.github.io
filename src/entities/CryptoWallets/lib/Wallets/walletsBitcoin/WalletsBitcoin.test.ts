import { suite, test, expect } from "vitest";

import WalletsBitcoin from "./WalletsBitcoin";
import type {
  BitcoinAddressFormat,
  BitcoinPrivateKeyFormat,
  BitcoinWalletPayload,
} from "@/entities/CryptoWallets/lib/Wallets/walletsBitcoin/WalletsBitcoin.types";

const key1 = new Uint8Array([
  132, 205, 117, 44, 180, 113, 245, 200, 213, 19, 97, 187, 70, 157, 86, 111, 19,
  131, 14, 18, 196, 57, 9, 203, 64, 196, 19, 26, 73, 35, 126, 80,
]);
const key2 = new Uint8Array([
  133, 205, 117, 44, 180, 113, 245, 200, 213, 19, 97, 187, 70, 157, 86, 111, 19,
  131, 14, 18, 196, 57, 9, 203, 64, 196, 19, 26, 73, 35, 126, 80,
]);

const notValidKey1 = new Uint8Array([
  255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
  255, 19, 131, 14, 18, 196, 57, 9, 203, 64, 196, 19, 26, 73, 35, 126, 80,
]);

const notValidKey2 = new Uint8Array([
  1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
  1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
  1, 1, 1, 1, 1, 1, 1,
]);

const variants: BitcoinWalletPayload[] = [];
for (const formatAddress of ["tr", "pkh", "wpkh"] as BitcoinAddressFormat[]) {
  for (const formatPrivateKey of [
    "hex",
    "array",
    "mainnet",
    "testnet",
    "mainnet-uncompressed",
    "testnet-uncompressed",
  ] as BitcoinPrivateKeyFormat[]) {
    for (const isTestnet of [true, false]) {
      if (formatPrivateKey === "mainnet" && isTestnet) continue;
      if (formatPrivateKey === "mainnet-uncompressed" && isTestnet) continue;
      if (formatPrivateKey === "testnet" && !isTestnet) continue;
      if (formatPrivateKey === "testnet-uncompressed" && !isTestnet) continue;
      variants.push({
        formatAddress,
        formatPrivateKey,
        isTestnet,
      });
    }
  }
}

suite("WalletsBitcoin", () => {
  test.each(variants)(
    "should generate wallet correctly with payload: %o",
    async (payload) => {
      const walletGenerator = new WalletsBitcoin();
      const wallet1 = await walletGenerator.makeWallet(payload, key1);
      expect(JSON.stringify(wallet1)).toMatchSnapshot();

      const wallet2 = await walletGenerator.makeWallet(payload, key2);
      expect(JSON.stringify(wallet2)).toMatchSnapshot();
    },
  );

  test("should generate wallet correctly without initialPrivateKey", async () => {
    const walletGenerator = new WalletsBitcoin();

    const payload: BitcoinWalletPayload = {
      formatPrivateKey: "hex",
      formatAddress: "pkh",
      isTestnet: true,
    };

    const wallet = await walletGenerator.makeWallet(payload);

    expect(typeof wallet.address).toBe("string");
    expect(wallet.address.length).not.toBe(0);
    expect(typeof wallet.privateKey).toBe("string");
    expect(wallet.privateKey.length).not.toBe(0);
  });

  test("should throw error for invalid initialPrivateKey", async () => {
    const walletGenerator = new WalletsBitcoin();

    const payload: BitcoinWalletPayload = {
      formatPrivateKey: "hex",
      formatAddress: "pkh",
      isTestnet: true,
    };

    await expect(
      walletGenerator.makeWallet(payload, notValidKey1),
    ).rejects.toThrow("Invalid private key");
    await expect(
      walletGenerator.makeWallet(payload, notValidKey2),
    ).rejects.toThrow("offset is out of bounds");
  });
});
