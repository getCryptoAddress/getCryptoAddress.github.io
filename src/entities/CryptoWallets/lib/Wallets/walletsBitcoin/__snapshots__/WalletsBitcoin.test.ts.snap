// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WalletsBitcoin > should generate wallet correctly with payload: {
  formatAddress: 'pkh',
  formatPrivateKey: 'mainnet-uncompressed',
  isTestnet: false
} 1`] = `"{\\"privateKey\\":\\"5JpmqxVfUkujA3LbVUZqqKjhUTUQVKHnzCob2Lv3qsTv3Vwc6fX\\",\\"address\\":\\"1BvrfZfJQQisjzEwWxQHiAMm8b9xCrGn8c\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: {
  formatAddress: 'pkh',
  formatPrivateKey: 'mainnet-uncompressed',
  isTestnet: false
} 2`] = `"{\\"privateKey\\":\\"5JqDPVMqTN87dwQaX3Ki8XpNAYFrhvZquQ3spAXYpWYSwoT78oT\\",\\"address\\":\\"133eaAGCnPwPF54o4i37DyNLWgLjgrt96K\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: {
  formatAddress: 'pkh',
  formatPrivateKey: 'testnet-uncompressed',
  isTestnet: true
} 1`] = `"{\\"privateKey\\":\\"92bQRhKD4yys86qt7pTkhvHf87q7eUpzL9fY6yGZBcCxpZPvMn5\\",\\"address\\":\\"mrSoxckHDSA8X6iZEXNfY5a5zakf7rvdFU\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: {
  formatAddress: 'pkh',
  formatPrivateKey: 'testnet-uncompressed',
  isTestnet: true
} 2`] = `"{\\"privateKey\\":\\"92bqyEBP3bCFbzus9PDd18NKpCcZs673FLuptnt4AFHViqoxxEx\\",\\"address\\":\\"mhZbsDMBbRNe2BYQnH1V3tafNfwSY2AcCK\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: {
  formatAddress: 'tr',
  formatPrivateKey: 'mainnet-uncompressed',
  isTestnet: false
} 1`] = `"{\\"privateKey\\":\\"5JpmqxVfUkujA3LbVUZqqKjhUTUQVKHnzCob2Lv3qsTv3Vwc6fX\\",\\"address\\":\\"bc1pgfd8hvk80u8ys3yphs48wfa8a7rldjckhatcs5dhk0a9q7dr7zqsmknvf0\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: {
  formatAddress: 'tr',
  formatPrivateKey: 'mainnet-uncompressed',
  isTestnet: false
} 2`] = `"{\\"privateKey\\":\\"5JqDPVMqTN87dwQaX3Ki8XpNAYFrhvZquQ3spAXYpWYSwoT78oT\\",\\"address\\":\\"bc1pflrar9kaegyee05fa7d5r3jqufl22q03xd6hmyxsr89s7dfrmkvslna73e\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: {
  formatAddress: 'tr',
  formatPrivateKey: 'testnet-uncompressed',
  isTestnet: true
} 1`] = `"{\\"privateKey\\":\\"92bQRhKD4yys86qt7pTkhvHf87q7eUpzL9fY6yGZBcCxpZPvMn5\\",\\"address\\":\\"tb1pgfd8hvk80u8ys3yphs48wfa8a7rldjckhatcs5dhk0a9q7dr7zqsv79rnq\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: {
  formatAddress: 'tr',
  formatPrivateKey: 'testnet-uncompressed',
  isTestnet: true
} 2`] = `"{\\"privateKey\\":\\"92bqyEBP3bCFbzus9PDd18NKpCcZs673FLuptnt4AFHViqoxxEx\\",\\"address\\":\\"tb1pflrar9kaegyee05fa7d5r3jqufl22q03xd6hmyxsr89s7dfrmkvsgmt3tk\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: {
  formatAddress: 'wpkh',
  formatPrivateKey: 'mainnet',
  isTestnet: false
} 1`] = `"{\\"privateKey\\":\\"L1frw5wWqWoj4uKVEgfxYHhTeBqxfyj7yV6dAvu9Yhjo816Es5PS\\",\\"address\\":\\"bc1qwlsvxyz08qe6h5pxe9q0n2ucwy8d3zzs3j53eh\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: {
  formatAddress: 'wpkh',
  formatPrivateKey: 'mainnet',
  isTestnet: false
} 2`] = `"{\\"privateKey\\":\\"L1hogGAxsNEQJ394wbFaYF24R6xaPdRxrGjhHB3qShPoWVg4XUCk\\",\\"address\\":\\"bc1qzec8yczhjvsnhfhd3f6lgsqzehnv86r6uqjqzp\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: {
  formatAddress: 'wpkh',
  formatPrivateKey: 'mainnet-uncompressed',
  isTestnet: false
} 1`] = `"{\\"privateKey\\":\\"5JpmqxVfUkujA3LbVUZqqKjhUTUQVKHnzCob2Lv3qsTv3Vwc6fX\\",\\"address\\":\\"bc1qwlsvxyz08qe6h5pxe9q0n2ucwy8d3zzs3j53eh\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: {
  formatAddress: 'wpkh',
  formatPrivateKey: 'mainnet-uncompressed',
  isTestnet: false
} 2`] = `"{\\"privateKey\\":\\"5JqDPVMqTN87dwQaX3Ki8XpNAYFrhvZquQ3spAXYpWYSwoT78oT\\",\\"address\\":\\"bc1qzec8yczhjvsnhfhd3f6lgsqzehnv86r6uqjqzp\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: {
  formatAddress: 'wpkh',
  formatPrivateKey: 'testnet-uncompressed',
  isTestnet: true
} 1`] = `"{\\"privateKey\\":\\"92bQRhKD4yys86qt7pTkhvHf87q7eUpzL9fY6yGZBcCxpZPvMn5\\",\\"address\\":\\"tb1qwlsvxyz08qe6h5pxe9q0n2ucwy8d3zzsm50zzy\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: {
  formatAddress: 'wpkh',
  formatPrivateKey: 'testnet-uncompressed',
  isTestnet: true
} 2`] = `"{\\"privateKey\\":\\"92bqyEBP3bCFbzus9PDd18NKpCcZs673FLuptnt4AFHViqoxxEx\\",\\"address\\":\\"tb1qzec8yczhjvsnhfhd3f6lgsqzehnv86r6kxfnej\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'pkh', formatPrivateKey: 'array', isTestnet: false } 1`] = `"{\\"privateKey\\":\\"132,205,117,44,180,113,245,200,213,19,97,187,70,157,86,111,19,131,14,18,196,57,9,203,64,196,19,26,73,35,126,80\\",\\"address\\":\\"1BvrfZfJQQisjzEwWxQHiAMm8b9xCrGn8c\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'pkh', formatPrivateKey: 'array', isTestnet: false } 2`] = `"{\\"privateKey\\":\\"133,205,117,44,180,113,245,200,213,19,97,187,70,157,86,111,19,131,14,18,196,57,9,203,64,196,19,26,73,35,126,80\\",\\"address\\":\\"133eaAGCnPwPF54o4i37DyNLWgLjgrt96K\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'pkh', formatPrivateKey: 'array', isTestnet: true } 1`] = `"{\\"privateKey\\":\\"132,205,117,44,180,113,245,200,213,19,97,187,70,157,86,111,19,131,14,18,196,57,9,203,64,196,19,26,73,35,126,80\\",\\"address\\":\\"mrSoxckHDSA8X6iZEXNfY5a5zakf7rvdFU\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'pkh', formatPrivateKey: 'array', isTestnet: true } 2`] = `"{\\"privateKey\\":\\"133,205,117,44,180,113,245,200,213,19,97,187,70,157,86,111,19,131,14,18,196,57,9,203,64,196,19,26,73,35,126,80\\",\\"address\\":\\"mhZbsDMBbRNe2BYQnH1V3tafNfwSY2AcCK\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'pkh', formatPrivateKey: 'hex', isTestnet: false } 1`] = `"{\\"privateKey\\":\\"84cd752cb471f5c8d51361bb469d566f13830e12c43909cb40c4131a49237e50\\",\\"address\\":\\"1BvrfZfJQQisjzEwWxQHiAMm8b9xCrGn8c\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'pkh', formatPrivateKey: 'hex', isTestnet: false } 2`] = `"{\\"privateKey\\":\\"85cd752cb471f5c8d51361bb469d566f13830e12c43909cb40c4131a49237e50\\",\\"address\\":\\"133eaAGCnPwPF54o4i37DyNLWgLjgrt96K\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'pkh', formatPrivateKey: 'hex', isTestnet: true } 1`] = `"{\\"privateKey\\":\\"84cd752cb471f5c8d51361bb469d566f13830e12c43909cb40c4131a49237e50\\",\\"address\\":\\"mrSoxckHDSA8X6iZEXNfY5a5zakf7rvdFU\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'pkh', formatPrivateKey: 'hex', isTestnet: true } 2`] = `"{\\"privateKey\\":\\"85cd752cb471f5c8d51361bb469d566f13830e12c43909cb40c4131a49237e50\\",\\"address\\":\\"mhZbsDMBbRNe2BYQnH1V3tafNfwSY2AcCK\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'pkh', formatPrivateKey: 'mainnet', isTestnet: false } 1`] = `"{\\"privateKey\\":\\"L1frw5wWqWoj4uKVEgfxYHhTeBqxfyj7yV6dAvu9Yhjo816Es5PS\\",\\"address\\":\\"1BvrfZfJQQisjzEwWxQHiAMm8b9xCrGn8c\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'pkh', formatPrivateKey: 'mainnet', isTestnet: false } 2`] = `"{\\"privateKey\\":\\"L1hogGAxsNEQJ394wbFaYF24R6xaPdRxrGjhHB3qShPoWVg4XUCk\\",\\"address\\":\\"133eaAGCnPwPF54o4i37DyNLWgLjgrt96K\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'pkh', formatPrivateKey: 'testnet', isTestnet: true } 1`] = `"{\\"privateKey\\":\\"cS2rPzwNGaVzELnkd6V5ucCXGR9NLRpp3XF6HMMf3pPoNk79beKj\\",\\"address\\":\\"mrSoxckHDSA8X6iZEXNfY5a5zakf7rvdFU\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'pkh', formatPrivateKey: 'testnet', isTestnet: true } 2`] = `"{\\"privateKey\\":\\"cS4o9BApJRvfTUcLL14huZX83LFz45XevJtAPbWLwp3omEnmXVXU\\",\\"address\\":\\"mhZbsDMBbRNe2BYQnH1V3tafNfwSY2AcCK\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'tr', formatPrivateKey: 'array', isTestnet: false } 1`] = `"{\\"privateKey\\":\\"132,205,117,44,180,113,245,200,213,19,97,187,70,157,86,111,19,131,14,18,196,57,9,203,64,196,19,26,73,35,126,80\\",\\"address\\":\\"bc1pgfd8hvk80u8ys3yphs48wfa8a7rldjckhatcs5dhk0a9q7dr7zqsmknvf0\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'tr', formatPrivateKey: 'array', isTestnet: false } 2`] = `"{\\"privateKey\\":\\"133,205,117,44,180,113,245,200,213,19,97,187,70,157,86,111,19,131,14,18,196,57,9,203,64,196,19,26,73,35,126,80\\",\\"address\\":\\"bc1pflrar9kaegyee05fa7d5r3jqufl22q03xd6hmyxsr89s7dfrmkvslna73e\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'tr', formatPrivateKey: 'array', isTestnet: true } 1`] = `"{\\"privateKey\\":\\"132,205,117,44,180,113,245,200,213,19,97,187,70,157,86,111,19,131,14,18,196,57,9,203,64,196,19,26,73,35,126,80\\",\\"address\\":\\"tb1pgfd8hvk80u8ys3yphs48wfa8a7rldjckhatcs5dhk0a9q7dr7zqsv79rnq\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'tr', formatPrivateKey: 'array', isTestnet: true } 2`] = `"{\\"privateKey\\":\\"133,205,117,44,180,113,245,200,213,19,97,187,70,157,86,111,19,131,14,18,196,57,9,203,64,196,19,26,73,35,126,80\\",\\"address\\":\\"tb1pflrar9kaegyee05fa7d5r3jqufl22q03xd6hmyxsr89s7dfrmkvsgmt3tk\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'tr', formatPrivateKey: 'hex', isTestnet: false } 1`] = `"{\\"privateKey\\":\\"84cd752cb471f5c8d51361bb469d566f13830e12c43909cb40c4131a49237e50\\",\\"address\\":\\"bc1pgfd8hvk80u8ys3yphs48wfa8a7rldjckhatcs5dhk0a9q7dr7zqsmknvf0\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'tr', formatPrivateKey: 'hex', isTestnet: false } 2`] = `"{\\"privateKey\\":\\"85cd752cb471f5c8d51361bb469d566f13830e12c43909cb40c4131a49237e50\\",\\"address\\":\\"bc1pflrar9kaegyee05fa7d5r3jqufl22q03xd6hmyxsr89s7dfrmkvslna73e\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'tr', formatPrivateKey: 'hex', isTestnet: true } 1`] = `"{\\"privateKey\\":\\"84cd752cb471f5c8d51361bb469d566f13830e12c43909cb40c4131a49237e50\\",\\"address\\":\\"tb1pgfd8hvk80u8ys3yphs48wfa8a7rldjckhatcs5dhk0a9q7dr7zqsv79rnq\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'tr', formatPrivateKey: 'hex', isTestnet: true } 2`] = `"{\\"privateKey\\":\\"85cd752cb471f5c8d51361bb469d566f13830e12c43909cb40c4131a49237e50\\",\\"address\\":\\"tb1pflrar9kaegyee05fa7d5r3jqufl22q03xd6hmyxsr89s7dfrmkvsgmt3tk\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'tr', formatPrivateKey: 'mainnet', isTestnet: false } 1`] = `"{\\"privateKey\\":\\"L1frw5wWqWoj4uKVEgfxYHhTeBqxfyj7yV6dAvu9Yhjo816Es5PS\\",\\"address\\":\\"bc1pgfd8hvk80u8ys3yphs48wfa8a7rldjckhatcs5dhk0a9q7dr7zqsmknvf0\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'tr', formatPrivateKey: 'mainnet', isTestnet: false } 2`] = `"{\\"privateKey\\":\\"L1hogGAxsNEQJ394wbFaYF24R6xaPdRxrGjhHB3qShPoWVg4XUCk\\",\\"address\\":\\"bc1pflrar9kaegyee05fa7d5r3jqufl22q03xd6hmyxsr89s7dfrmkvslna73e\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'tr', formatPrivateKey: 'testnet', isTestnet: true } 1`] = `"{\\"privateKey\\":\\"cS2rPzwNGaVzELnkd6V5ucCXGR9NLRpp3XF6HMMf3pPoNk79beKj\\",\\"address\\":\\"tb1pgfd8hvk80u8ys3yphs48wfa8a7rldjckhatcs5dhk0a9q7dr7zqsv79rnq\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'tr', formatPrivateKey: 'testnet', isTestnet: true } 2`] = `"{\\"privateKey\\":\\"cS4o9BApJRvfTUcLL14huZX83LFz45XevJtAPbWLwp3omEnmXVXU\\",\\"address\\":\\"tb1pflrar9kaegyee05fa7d5r3jqufl22q03xd6hmyxsr89s7dfrmkvsgmt3tk\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'wpkh', formatPrivateKey: 'array', isTestnet: false } 1`] = `"{\\"privateKey\\":\\"132,205,117,44,180,113,245,200,213,19,97,187,70,157,86,111,19,131,14,18,196,57,9,203,64,196,19,26,73,35,126,80\\",\\"address\\":\\"bc1qwlsvxyz08qe6h5pxe9q0n2ucwy8d3zzs3j53eh\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'wpkh', formatPrivateKey: 'array', isTestnet: false } 2`] = `"{\\"privateKey\\":\\"133,205,117,44,180,113,245,200,213,19,97,187,70,157,86,111,19,131,14,18,196,57,9,203,64,196,19,26,73,35,126,80\\",\\"address\\":\\"bc1qzec8yczhjvsnhfhd3f6lgsqzehnv86r6uqjqzp\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'wpkh', formatPrivateKey: 'array', isTestnet: true } 1`] = `"{\\"privateKey\\":\\"132,205,117,44,180,113,245,200,213,19,97,187,70,157,86,111,19,131,14,18,196,57,9,203,64,196,19,26,73,35,126,80\\",\\"address\\":\\"tb1qwlsvxyz08qe6h5pxe9q0n2ucwy8d3zzsm50zzy\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'wpkh', formatPrivateKey: 'array', isTestnet: true } 2`] = `"{\\"privateKey\\":\\"133,205,117,44,180,113,245,200,213,19,97,187,70,157,86,111,19,131,14,18,196,57,9,203,64,196,19,26,73,35,126,80\\",\\"address\\":\\"tb1qzec8yczhjvsnhfhd3f6lgsqzehnv86r6kxfnej\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'wpkh', formatPrivateKey: 'hex', isTestnet: false } 1`] = `"{\\"privateKey\\":\\"84cd752cb471f5c8d51361bb469d566f13830e12c43909cb40c4131a49237e50\\",\\"address\\":\\"bc1qwlsvxyz08qe6h5pxe9q0n2ucwy8d3zzs3j53eh\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'wpkh', formatPrivateKey: 'hex', isTestnet: false } 2`] = `"{\\"privateKey\\":\\"85cd752cb471f5c8d51361bb469d566f13830e12c43909cb40c4131a49237e50\\",\\"address\\":\\"bc1qzec8yczhjvsnhfhd3f6lgsqzehnv86r6uqjqzp\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'wpkh', formatPrivateKey: 'hex', isTestnet: true } 1`] = `"{\\"privateKey\\":\\"84cd752cb471f5c8d51361bb469d566f13830e12c43909cb40c4131a49237e50\\",\\"address\\":\\"tb1qwlsvxyz08qe6h5pxe9q0n2ucwy8d3zzsm50zzy\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'wpkh', formatPrivateKey: 'hex', isTestnet: true } 2`] = `"{\\"privateKey\\":\\"85cd752cb471f5c8d51361bb469d566f13830e12c43909cb40c4131a49237e50\\",\\"address\\":\\"tb1qzec8yczhjvsnhfhd3f6lgsqzehnv86r6kxfnej\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'wpkh', formatPrivateKey: 'testnet', isTestnet: true } 1`] = `"{\\"privateKey\\":\\"cS2rPzwNGaVzELnkd6V5ucCXGR9NLRpp3XF6HMMf3pPoNk79beKj\\",\\"address\\":\\"tb1qwlsvxyz08qe6h5pxe9q0n2ucwy8d3zzsm50zzy\\"}"`;

exports[`WalletsBitcoin > should generate wallet correctly with payload: { formatAddress: 'wpkh', formatPrivateKey: 'testnet', isTestnet: true } 2`] = `"{\\"privateKey\\":\\"cS4o9BApJRvfTUcLL14huZX83LFz45XevJtAPbWLwp3omEnmXVXU\\",\\"address\\":\\"tb1qzec8yczhjvsnhfhd3f6lgsqzehnv86r6kxfnej\\"}"`;
